src/Tribe/API.php

```diff
@@ -602,5 +642,222 @@ public static function get_and_flatten_event_meta( $event_id ) {
+			) {
+				$args['EventEndMeridian'] = '';
+			}
+
+			// Now process all but the meridians
+			foreach ( $args as $key => &$value ) {
+				if ( 'EventStartHour' === $key ) {
+					if ( empty( $args['EventStartMeridian'] ) ) {
+						$value = self::sanitize_hour_meta_value( $value );
+					} else {
+						$value = self::sanitize_hour_meta_value( $value, true );
+					}
+				} elseif ( 'EventEndHour' === $key ) {
+					if ( empty( $args['EventEndMeridian'] ) ) {
+						$value = self::sanitize_hour_meta_value( $value );
+					} else {
+						$value = self::sanitize_hour_meta_value( $value, true );
+					}
+				} elseif (
+					'EventStartMinute' === $key
+					|| 'EventEndMinute' === $key
+				) {
+					$value = self::sanitize_minute_meta_value( $value );
+				}
+
+				if ( is_wp_error( $value ) ) {
+					return $value;
+				}
+			}
+
+			return $args;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event meridian post meta value: am|pm.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value            The post meta value to be checked, such as 'am'.
+		 * @param bool   $empty_if_invalid If true, set an invalid value to an empty string, else generate WP_Error.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_meridian_meta_value( $value, $empty_if_invalid = true ) {
+			// lower-case to match the `a` PHP date format used elsewhere
+			$new_value = strtolower( trim( $value ) );
+
+			if (
+				'am' !== $new_value
+				&& 'pm' !== $new_value
+			) {
+				if ( $empty_if_invalid ) {
+					$new_value = '';
+				} else {
+					$error_message = sprintf(
+						esc_html__( 'An event having a post meta value of `%s` meridian is not valid. Make sure it is either `am` or `pm`, or remove it entirely if using the 24-hour format.', 'the-events-calendar' ),
+						$value
+					);
+
+					return new WP_Error( 'invalid-tribe-events-meridian-meta-value', $error_message );
+				}
+			}
+
+			return $new_value;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event hour post meta value: 1-12 if `Event{Start|End}Meridian`
+		 * is also passed, else 0-23.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value       The post meta value to be checked, such as '07'.
+		 * @param bool   $twelve_hour If false, allow 0-23. If true, only allow 1-12.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_hour_meta_value( $value, $twelve_hour = false ) {
+			$value = (string) $value;
+
+			if ( ! is_numeric( $value ) ) {
+				$error_message = sprintf(
+					esc_html__( 'An event having a post meta value of `%s` hour (am/pm) is not valid. Make sure it is from 1 to 12.', 'the-events-calendar' ),
+					$value
+				);
+
+				return new WP_Error( 'non-numeric-tribe-events-hour-meta-value', $error_message );
+			}
+
+			$new_value = absint( $value );
+
+			if ( $twelve_hour ) {
+				if (
```

Comment:
I think this conditional can fit in a single line.

Reply:
Also I think by inverting the conditional here make it more easier to follow see the details commented on `sanitize_minute_meta_value`

Reply:
If `$twelve_hour` is `true` but the hour is `13` -- and the conditional was all together -- it would failover to 24-hour format conditional, which isn't correct. Please let me know if this is incorrect.

Reply:
oh, no, incorrect... will fix